### ISSUE TITLE
wk4: Reduce file size for gifs

### ISSUE THREAD

**Issue by ritahitching** [lzim/teampsd#1150]
**Research options to zip gifs** 

Research options to zip/reduce the size of gifs. 
Estimate 2 hour.

**Comment by ritahitching** [comment]
@lzim i need some guidance on reducing the size of the gifs, 

1. online platform to zip the gifs https://ezgif.com/optimize 
2. size reduction (e.g. 1MB to 743KB)
3. gifs attached

Please let me know if resolution ok, and if I should continue with the remainder of the gifs [here](https://github.com/lzim/teampsd/tree/master/resources/gifs/mtl_2.0)

**original 1MB**
![community_button_1mb](https://user-images.githubusercontent.com/54862187/76442481-400f6000-637e-11ea-8d7c-ceea70c58fff.gif)

**reduced 743KB**
![community_button_743k](https://user-images.githubusercontent.com/54862187/76442552-5b7a6b00-637e-11ea-81b8-abe10ae6ec1d.gif)

**Comment by staceypark** [comment]
Two points of feedback:
- resolution looks fine
- is there any way to make this size even smaller? I'm wondering if this reduction is enough

**Comment by anthonycpichardo** [comment]
Resolution looks good to me.

I found this useful article on different ways to improve gif related performance if you want to scroll thing. One hot tip seemed to be converting from gif to html5.
https://moz.com/blog/how-to-improve-site-performance-using-gifs

**Comment by lzim** [comment]
FYI @ritahitching 
I’m catching up on the weekend, but this can definitely wait until Monday 😊

@anthonycpichardo  and @staceypark thanks for the feedback 

Let’s get more testing done following the animation instructions in Anthony’s post.

Specifically: 
1. **Lossy optimization** Produce an an example, _AND/OR_ report the specific details to how the example above was compressed 
2. **HTML5** Produce an example as described in Anthony’s and post here for review

**Comment by ritahitching** [comment]
**Large file** I chose a large file to stress test the compression process.

**Lossy** Optimization using **https://ezgif.com/optimize**

**1. there are several options for several degrees of compression from light to heavy (30 to 200)**
![Screen Shot 2020-03-16 at 3 56 14 PM](https://user-images.githubusercontent.com/54862187/76807272-10f35700-67a1-11ea-947c-b730186b9066.png)



**2. original image - File size: 3.2MiB, width: 800px, height: 478px, frames: 216, type: gif**

![data_ui_splashpage_original](https://user-images.githubusercontent.com/54862187/76807323-3718f700-67a1-11ea-98ff-c35d1febf09f.gif)


**3. shown 3 versions below**

**35 compression: File size: 1.92MiB (-39.91%), width: 800px, height: 478px, frames: 215, type: gif** 
![data_ui_35](https://user-images.githubusercontent.com/54862187/76807492-c6bea580-67a1-11ea-8055-5b1e649b73f0.gif)



****90 compression: File size: 1.77MiB (-44.7%), width: 800px, height: 478px, frames: 215, type: gif**
![data_ui_90](https://user-images.githubusercontent.com/54862187/76807562-fd94bb80-67a1-11ea-8dda-3c73acce00bd.gif)



**160 compression: File size: 1.67MiB (-47.86%), width: 800px, height: 478px, frames: 215, type: gif**

![data_ui_160](https://user-images.githubusercontent.com/54862187/76807642-43518400-67a2-11ea-89c7-919ae24f7e6f.gif)

**Lossy GIF compression Details**
GIF compressor optimizes GIFs using gifsicle and Lossy GIF encoder, which implements lossy LZW compression.
It can reduce animated GIF file size by 30%—50% at a cost of some dithering/noise. You can adjust compression level with a simple slider to get the best result for your use case.
This is the default method and should work for any GIF.

**Color Reduction**
This tool shrinks GIF file size by reducing the number of colors in each frame.
Each GIF frame can use up to 256 unique colors, and by reducing this number, you can achieve a smaller file size.
It makes multiple variations of your input image, and you can choose the one that has the best size/quality ratio for your needs.

**Remove every nth frame**
The frame drop option can remove every second, third, or fourth frame to reduce frame rate and, therefore, file size. Useful for long gifs with a high frame rate.

Another option is to remove only duplicate frames. This method will search for identical or very similar, consecutive frames, remove them and merge their display duration.
This will not do anything for most GIFs, but can be useful in some special cases.

**Optimize Transparency**
This option keeps the first frame in the background, and makes unchanged parts of the following frames transparent.
The results heavily depends on the nature of the image. It may give huge file size reduction in some cases when images have large, static areas, e.g. recorded screen capture of some program, but most likely will give poor results for converted videos or photo slide shows, where all or most pixels between each frame are completely different. The fuzz factor represents how similar colors can be considered as equal.

**Comment by ritahitching** [comment]
**HTML5 Compression Using https://ezgif.com/optimize** 

**Conversion to HTML5**
Github does not support the upload of gifs converted to
1. HTML5
2. MP4 video

I can circulate images via email. Please let know.

**Comment by staceypark** [comment]
@anthonycpichardo @lzim @ritahitching 
I think the most compressed one still looks fine from Rita's newest examples.
But then again I don't think I have the best eye for these since I don't notice the fuzzy icons.. 🤣

**Comment by ritahitching** [comment]
@staceypark thanks for the feedback. 
@lzim and @anthonycpichardo thoughts? thanks

**Comment by ritahitching** [comment]
@lijenn @anazariz @anthonycpichardo will you please look at the gifs above using the LOSSY compression and let me know if they are clear to you. thanks

**Comment by ritahitching** [comment]
@lijenn @anazariz @anthonycpichardo  If you are able to look at the images in different monitors and send some screen shots it wold be really helpful.
thanks

**Comment by lijenn** [comment]
@ritahitching 
I think the original, 35, and 90 compression all look very similar. I do see a minor drop in quality for the 160 compression though.

Here are screenshots of what I'm seeing on my mac pro 13inch display:

Original:
<img width="802" alt="original" src="https://user-images.githubusercontent.com/59668647/77188173-d59fa380-6a92-11ea-9e75-92b582354a93.png">

35 Compression:
<img width="798" alt="1" src="https://user-images.githubusercontent.com/59668647/77188291-f8ca5300-6a92-11ea-98ec-5b2ed64bddc1.png">

90 Compression:
<img width="802" alt="2" src="https://user-images.githubusercontent.com/59668647/77188293-fa941680-6a92-11ea-897a-5f2e716f292c.png">

160 Compression:
<img width="799" alt="3" src="https://user-images.githubusercontent.com/59668647/77188302-fd8f0700-6a92-11ea-9038-5a6d97db62ad.png">

**Comment by ritahitching** [comment]
@anthonycpichardo and @lzim any decision on the choice and degree of compression for the gifs?
thanks!

**Comment by anthonycpichardo** [comment]
I agree with jenn, I dont really notice a drop until we get to 160.

**Comment by ritahitching** [comment]
@anthonycpichardo thanks for feedback.
@lzim will you please provide a final decision on the choice and degree of compression? 
Please let me know if I should go ahead with compression for all the gifs in the [MTL 2.0 folder](https://github.com/lzim/teampsd/tree/master/resources/gifs/mtl_2.0) once you've made you've made your decision. thanks!

**Comment by lijenn** [comment]
@ritahitching Let's go with the 90 compression since we have some consensus about 160 showing a drop in quality. 

Could you close this card when the task is completed? Thank you!

**Comment by ritahitching** [comment]
@ritahitching provide the team with guidance on how to use gifs for our purpose of instructional guides.

**Comment by lijenn** [comment]
From Mon. 3/30 WG Leads Meeting:
Can @branscombj & @dlkibbe provide a second set of eyes on these gifs? 

Please look at the examples and give feedback on whether or not the gifs can be followed through from the point of view as a learner (i.e is going through the motions enough in the gif or do you think it needs to be of higher quality to follow each step).

**Comment by ritahitching** [comment]
@branscombj if you a chance will you please review the at the gifs and let me know your thoughts? thanks!

**Comment by lzim** [comment]
@ritahitching 

- We are still awaiting an update on your research regarding best practices for .gif instruction.
- Not sure if these can be slowed down in any way as part of finalizing our compression plans.

@branscombj Apologizes @lijenn already remade all the .gifs and now we are simply trying to reduce the total file size so that pages on slow machines with limited RAM (like my VA machine) can keep up.

**Comment by ritahitching** [comment]
**from @ritahitching** 
1. The recommended format for gif construction is [jpg](https://themeisle.com/blog/best-image-format/) for gifs
### Summary:
You can optimize GIF files, but in most cases, the results won’t be as good as with other image formats. That means you should aim to use GIFs sparingly throughout your website, only when you need to showcase an animation you can’t create any other way (such as with CSS). On top of that, GIFs only support up to 256 colors, as well as transparency. That makes them a poor choice for displaying complex images.

2. The recommended format for [gif compression](https://devcodegeek.com/best-gifs-compression-tools.html)
### Summary:
[GIFReducer](http://www.gifreducer.com/) reduces the several colors from the GIF image or file. It only works to make the smaller files. But it can improve the quality of an image. Better than  previously recommended **Lossy** compression mode reduces both size plus quality of the image.

3. I could not find any information about slowing down the display of the gifs. 

from @lzim 
- We are still awaiting an update on your research regarding best practices for .gif instruction.
- Not sure if these can be slowed down in any way as part of finalizing our compression plans.

**Comment by staceypark** [comment]
I like the idea of screenshots to complement critical steps. Esp since it also lets you highlight or circle specific sections.

> @ritahitching @lzim
> I believe I could try to slow them down with any gif editor out on the web. However, as mentioned, this may affect the file size and our plan to reduce it as much as possible and also hurt the quality even more as changes are being made on it.
> 
> Maybe the addition of screenshots could address the issue of needing to look more carefully at drop-down menus or areas with a lot of text involved? Screenshots have a much smaller file size and are static which also addresses the issue of not being able to "pause" a gif, a point @branscombj brought up earlier in our WG meeting.

**Comment by ritahitching** [comment]
@staceypark @lijenn 
- [ ] Does it make sense to for this issue to be triaged at our HQ meeting 4/9 so that we can make a decision on the next steps? 
- [ ] If yes, I'll update the agenda accordingly.


> I like the idea of screenshots to complement critical steps. Esp since it also lets you highlight or circle specific sections.
> 
> > @ritahitching @lzim
> > I believe I could try to slow them down with any gif editor out on the web. However, as mentioned, this may affect the file size and our plan to reduce it as much as possible and also hurt the quality even more as changes are being made on it.
> > Maybe the addition of screenshots could address the issue of needing to look more carefully at drop-down menus or areas with a lot of text involved? Screenshots have a much smaller file size and are static which also addresses the issue of not being able to "pause" a gif, a point @branscombj brought up earlier in our WG meeting.

**Comment by ritahitching** [comment]
@lzim I need some decisions on this issue:

**1. Based on the research I provided above on gif compression**:
- [ ] Are you ok for me to go ahead and compress the images using the recommendations in the article by [gif compression](https://devcodegeek.com/best-gifs-compression-tools.html) to use [GIFReducer](http://www.gifreducer.com/)?
- [ ] The article  suggests the compression algorithm offered by [GIFReducer](http://www.gifreducer.com/) for gifs as it reduces the size of the files without loss to image quality. The previously recommended **Lossy** compression reduces gif size and quality.

**2. Speed at which gifs are displayed**
- [ ] Decision on cost/benefit of slowing down the presentation of the gifs - @lijenn believes it's possible (see her comment below) but the issue of large files sizes remains. 
- [ ] Decision on suggestion by @staceypark and @lijenn to compressed gifs to reduce loading lag but not change the display speed of the gifs . Proposal to add screenshots for critical steps (which are naturally smaller files, yet retain clear details, without increasing loading lag).
 

**from @branscombj **
 I do feel that the gif moves too fast for my brain to keep up. It would help me to have more time on the drop-down menu, and more time to catch the spreadsheet column headings and the chart title, axes, and legend to be able to make the connection between what's being selected and what's being shown. I'm not sure it's necessary to show the scrolling down to include the file folders unless you want to point something out about them.

**from @lijenn ** 
I believe I could try to slow them down with any gif editor out on the web. However, as mentioned, this may affect the file size and our plan to reduce it as much as possible and also hurt the quality even more as changes are being made on it. 

Maybe the addition of screenshots could address the issue of needing to look more carefully at drop-down menus or areas with a lot of text involved? Screenshots have a much smaller file size and are static which also addresses the issue of not being able to "pause" a gif, a point @branscombj brought up earlier in our WG meeting. 

**from @staceypark **
I like the idea of screenshots to complement critical steps. Esp since it also lets you highlight or circle specific sections.

**Comment by ritahitching** [comment]
@lzim I'm sorry to ping you again :pray: 
I know you are slammed :muscle: with the grant submission, but are you able to make a decision on the choice of file compression? thank you

**Comment by lzim** [comment]
@ritahitching 

1. The .gifs are definitely not a bigger priority than the COVID-19 grants and updated compliance tasks. I have adjusted the **title**, **milestone,** and **epic** to accurate reflect May.
- _This is great, it means there's more capacity to help with higher priority **april** epic tasks._ 🧯 
---
2. I'm not sure what your recommendations are from the group based on the details above.
**Can you concisely describe your recommendations, please?**
- For example, is @branscombj's feedback specific to this .gif only, or is it a generalizable comment? Can you "active listen" to see what inferences you should draw?
---
3. @lijenn and @staceypark I notice that sometimes having a lot of screenshots and .gifs actually interferes with my ability to just read what a guide says (e.g. https://github.com/lzim/mtl/blob/master/blue/session02/s02_learner/mtl_session02_see.md)
- This is especially problematic when differing dimensions are used so they don't align as you scroll down the page.
---
4. I'm really struggling to interpret the **estimate of 3 story points**? Really? That's all this would take?
- **Can you please consider a more accurate reflection of how much re-work this entails?
- We really need the accurate estimates to determine whether/when/how to do this.

**Comment by ritahitching** [comment]
@lzim thank you for the coaching.  Please see answers below.

- [ ] I agree. We have greater priorities than the gifs.
- [ ] As such, I just offered to take over #1294 and code the RedCap measures for the NCPTSD 10-item Coronavirus Stress measure and #1339 on th biosketches for the NIDA supplement.
- [ ] I believe in the SOP, it is Workgroup Leads that make decisions on moving an issue to another Epic.  I was not informed that this issue was being moved to the May Epic, but told to get a decision, start the work do the work and close this issue. I would not have asked about a decision on this issue, if I was aware that it was being moved to the May Epic Thank you for updating the issue to reflect the change.

**from @lzim** 
1. The .gifs are definitely not a bigger priority than the COVID-19 grants and updated compliance tasks. I have adjusted the **title**, **milestone,** and **epic** to accurate reflect May.
- _This is great, it means there's more capacity to help with higher priority **april** epic tasks._ 🧯 
---

**from @ritahitching** 
- [ ] I recommend the compression algorithm offered by [GIFReducer](http://www.gifreducer.com/) for gifs  -  it reduces the size of the files without loss to image quality. 
- [ ] My interpretation of @branscombj is that gifs are not her preference, as clarity is more important.
- [ ] I agree with your comment below, that too many gifs/screenshots makes it difficult to read the guide.

**from @lzim** 
2. I'm not sure what your recommendations are from the group based on the details above.
**Can you concisely describe your recommendations, please?**
- For example, is @branscombj's feedback specific to this .gif only, or is it a generalizable comment? Can you "active listen" to see what inferences you should draw?
---
3. @lijenn and @staceypark I notice that sometimes having a lot of screenshots and .gifs actually interferes with my ability to just read what a guide says (e.g. https://github.com/lzim/mtl/blob/master/blue/session02/s02_learner/mtl_session02_see.md)
- This is especially problematic when differing dimensions are used so they don't align as you scroll down the page.
---
**from @ritahitching** 
- [ ] The story points were to do the research on the appropriate choice of compression. 
- [ ] I was following the SOP of breaking down tasks into individual issue cards that can be closed.  I was planning a separate issue card to do the work of compressing the images.  
- [ ] The compression I would estimate at about 5 points.  Total for both research and work 8 points.   

**from @lzim** 
4. I'm really struggling to interpret the **estimate of 3 story points**? Really? That's all this would take?
- Can you please consider a more accurate reflection of how much re-work this entails?
- We really need the accurate estimates to determine whether/when/how to do this.

**Comment by ritahitching** [comment]
@lijenn this is issue is blocked by the #1192 and the MVP process.
If you read the thread you will find recommended suggestions on the choice of file compression.  No decisions were made.
I recommend we keep this issue open.

**Comment by lijenn** [comment]
File sizes reduced, may need to reduce new S2 data gifs for facility and data ui for VVC updates in Dec. epic.
